[release/9.0] Update Microsoft.Build.* versions to 17.8.43

PranavSenthilnathan opened the pull request:
Backport of https://github.com/dotnet/runtime/pull/120818

PranavSenthilnathan committed:
[release/9.0] Update Microsoft.Build.* versions to 17.8.43

ViktorHofer commented:
This means that you either need to include those packages in 9.0 SBRP or make it use the N-1 version. @dotnet/source-build can you help you further with this.

ellahathaway commented:
> This means that you either need to include those packages in 9.0 SBRP or make it use the N-1 version. @dotnet/source-build can you help you further with this.

These packages will have to be added in 9.0 SBRP. SBRP is a self-service repo, so please follow the guidance [here](https://github.com/dotnet/source-build-reference-packages/tree/release/9.0?tab=readme-ov-file#adding-new-packages). You can use https://github.com/dotnet/source-build-reference-packages/pull/885 as a reference.

dotnet-maestro[bot] committed:
Update dependencies from https://github.com/dotnet/source-build-reference-packages build 20251017.3
On relative base path root
Microsoft.SourceBuild.Intermediate.source-build-reference-packages From Version 9.0.0-alpha.1.25506.2 -> To Version 9.0.0-alpha.1.25517.3

ellahathaway commented:
@PranavSenthilnathan - I've cherry-picked the commit from https://github.com/dotnet/runtime/pull/120854. The source-build leg should pass now :)

PranavSenthilnathan commented:
/ba-g #120577
